doc/Learn/Storage/HowTo-RequiredEntitlements.md

```diff
@@ -0,0 +1,64 @@
+---
+uid: Uno.Extensions.Storage.HowToRequiredEntitlements
+---
+# How-To: Adding Required Entitlements
+
+On Apple platforms (iOS, Mac Catalyst) the Uno storage extension use the OS Key Chain service to store secrets. This requires your application to have the [`keychain-access-groups`](https://developer.apple.com/documentation/bundleresources/entitlements/keychain-access-groups) entitlement to work properly.
+
+## Step-by-steps
+
+> [!IMPORTANT]
+> This guide assumes you used the template wizard or `dotnet new unoapp` to create your solution. If not, it is recommended that you follow the [instructions](xref:Uno.Extensions.HowToGettingStarted) for creating an application from the template.
+
+### 1. Adding the Entitlements.plist file
+
+The default location, inside your project, for the new file(s) should be:
+
+* iOS: `iOS/Entitlements.plist`
+* Mac Catalyst: `MacCatalyst/Entitlements.plist`
+
+The content of the file(s) should be:
+
+```xml
+<key>keychain-access-groups</key> 
+ <array> 
+ 	<string>$(AppIdentifierPrefix)$(CFBundleIdentifier)</string> 
+ </array>
+```
+
+For more information see Apple's documentation related to the [Key Chain](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps?language=objc).
+
+The variables `$(AppIdentifierPrefix)` and `$(CFBundleIdentifier)` will be replaced with the correct values at build time. For more information about how the Microsoft .NET SDK works with entitlements you can consult:
+
+* [Microsoft iOS](https://learn.microsoft.com/en-us/dotnet/maui/ios/entitlements)
+* [Microsoft Mac Catalyst](https://learn.microsoft.com/en-us/dotnet/maui/mac-catalyst/entitlements)
+
+### 2. Add capabilities in your Apple Developer Account
+
+Adding the `Entitlements.plist` to your project is not enough. You must also add the capability inside your Apple Developer Account and create a provisioning profile specific for your application. You can follow Microsoft's instruction for both steps:
+
+* [Microsoft iOS](https://learn.microsoft.com/en-us/dotnet/maui/ios/capabilities?#add-capabilities-in-your-apple-developer-account)
+* [Microsoft Mac Catalyst](https://learn.microsoft.com/en-us/dotnet/maui/mac-catalyst/capabilities?#add-capabilities-in-your-apple-developer-account)
+
+### 3. Modifying the `*.Mobile.csproj`
+
+A new property group should be added to your `*.csproj` project file. The example below will work for both iOS and Mac Catalyst targets.
+
+```xml
+<PropertyGroup>
+    <CodesignEntitlements Condition="$(IsIOS)">iOS\Entitlements.plist</CodesignEntitlements>
+    <CodesignEntitlements Condition="$(IsMacCatalyst)">MacCatalyst\Entitlements.plist</CodesignEntitlements>
+    <CodesignKey>Apple Development: Some User (XXXXXXXXXX)</CodesignKey>
+    <CodesignProvision>Mac Catalyst Team Provisioning Profile: com.companyname.maccatalyst</CodesignProvision>
+</PropertyGroup>
+```
```

Comment:
Perhaps make a note here that the CodesignKey and CodesignProvision are only required for catalyst, since iOS will apply these as part of the auto-provision

Reply:
> auto-provision

Hmm... not sure it's the right term. The SDK tooling tries to figure out an identity / PP for your app. If you have a few (like 1), very general identities and PP then it works fine. If you have many then you can get into different problems (at build or run time). 

IMO it's good advice (for app developers) to specify what to use - even more when a specific PP is (or will be) needed.